### PR TITLE
Add DaneeCloud to File Transfer - Object Storage & File Servers

### PR DESCRIPTION
**Pull request by DaneeSkripter**
Thank you for taking the time to work on a PR for Awesome-Selfhosted!

To ensure your PR is dealt with swiftly please check the following:

- [x] Submit one item per pull request. This eases reviewing and speeds up inclusion.
- [x] Format your submission as follows, where `Demo` and `Clients` are optional.
  Do not add a duplicate `Source code` link if it is the same as the main link.
  Keep the short description under 250 characters and use [sentence case](https://en.wikipedia.org/wiki/Letter_case#Sentence_case)
  for it, even if the project's webpage or readme uses another capitalisation
  such as title case, all caps, small caps or all lowercase.
  `Demo` links should only be used for interactive demos, i.e. not video demonstrations.
  ``- [Name](http://homepage/) - Short description, under 250 characters, sentence case. ([Demo](http://url.to/demo), [Source Code](http://url.of/source/code), [Clients](https://url.to/list/of/related/clients-or-apps)) `License` `Language` ``
- [x] Additions that depend on proprietary services outside the user's control must be marked `⚠`.
  ``- [Name](http://homepage/) `⚠` - Short description, under 250 characters, sentence case. ([Demo](http://url.to/demo), [Source Code](http://url.of/source/code), [Clients](https://url.to/list/of/related/clients-or-apps)) `License` `Language` ``
- [x] Additions that are not [Free software](https://en.wikipedia.org/wiki/Free_software) must be added to `non-free.md` and have the license set to `⊘ Proprietary`
- [x] Additions are inserted preserving alphabetical order.
- [x] Additions are not already listed at any of [awesome-sysadmin](https://github.com/n1trux/awesome-sysadmin), [staticgen.com](https://www.staticgen.com/), [staticsitegenerators.net](https://staticsitegenerators.net/), [dbdb.io](https://dbdb.io/browse).
- [x] The `Language` tag is the main **server-side** requirement for the software. Don't include frameworks or specific dialects.
- [x] Any license you add is in our [list of licenses](https://github.com/awesome-selfhosted/awesome-selfhosted/blob/master/README.md#list-of-licenses).
- [x] You have searched the repository for any relevant [issues](https://github.com/awesome-selfhosted/awesome-selfhosted/issues) or [PRs](https://github.com/awesome-selfhosted/awesome-selfhosted/pulls), including closed ones.
- [x] Any category you are creating has the minimum requirement of 3 items.
  If not, your addition may be inserted into `Misc/Other`.
- [x] Any software project you are adding to the list is actively maintained.
- [x] Any software project you are adding has working installation instructions.
- [x] The pull request title is informative, unlike "Update README.md".
  Suggested titles: "Add aaa to bbb" for adding software aaa to section bbb,
  "Remove aaa from bbb" for removing, "Fix license for aaa", etc.

### CODE REVIEW

**Comment by nodiscc** [comment]
@DaneeSkripter Please check the relevant boxes in the pull request template.

**Comment by DaneeSkripter** [comment]
> @DaneeSkripter Please check the relevant boxes in the pull request template.

Done!

**Review by kokomo123 — APPROVED** [review]
It looks all good from here, just some corrections with management instead of managment, thanks for your contribution!

**Comment by DaneeSkripter** [comment]
I fixed some grammar mistakes.

**Review by kokomo123 — APPROVED** [review]
Thank you for fixing it, looks perfect! @nodiscc I say this is ready for merging.

**Comment by DaneeSkripter** [comment]
@nodiscc Can you merge it please?

**Comment by nodiscc** [comment]
The description must be improved - it should explain what the software does.

**Comment by kokomo123** [comment]
I would've thought it looked alright, but yeah custom cloud service is kinda vague, my mistake.

**Comment by DaneeSkripter** [comment]
@kokomo123  @nodiscc  Does it look better now?

**Review by kokomo123 — COMMENTED** [review]
This is a better description for sure, and it  works as it conveys what the software does much better now. There's also the question of MongoDB, is it required to use MongoDB, or are there plans for people to use other databases?

**Comment by DaneeSkripter** [comment]
> This is a better description for sure, and it works as it conveys what the software does much better now. There's also the question of MongoDB, is it required to use MongoDB, or are there plans for people to use other databases?

MongoDB is required to use.

**Comment by kokomo123** [comment]
I'm not sure about this then, it's either we move it to non-free, or, it may warrant putting a `⚠` as MongoDB is proprietary. I shall await for advice from @nodiscc, so let's be patient until he gives advice.

**Comment by Rabenherz112** [comment]
MongoDB can be self-hosted as Community Edition, so I don't see how this is proprietary. Granted, 99% of users will probably use the cloud service since it is free and easier to set up but if you want, you can always host it by yourself.
Source Code: https://github.com/mongodb/mongo
Install Instructions for pre-build package: https://www.mongodb.com/docs/manual/tutorial/install-mongodb-on-debian/

**Comment by kokomo123** [comment]
It's SSPL, it isn't recognized as "free software" but the SSPL permits the source to be available. It's free to use, my apologies.

**Comment by Rabenherz112** [comment]
Ups, apologies for the oversight. I didn't notice that Mongo uses the SSPL License. Sorry for that..
I now share your concerns regarding the MongoDB requirement. Considering that the SSPL is categorized as a non-free (proprietary) license, and this particular database is the sole supported option, it would indeed necessitate the use of the proprietary dependency flag.

Especially since other Software which uses this licence is also listed under the non-free page. But I also agree that this should be something that nodiscc should double check on.


P.S.: If I have understood the license requirements correctly, doesn't the SSPL license primarily impose restrictions on cloud service providers, owing to its extended copyleft provisions?

**Comment by kokomo123** [comment]
> Ups, apologies for the oversight. I didn't notice that Mongo uses the SSPL License. Sorry for that.. I now understand your concerns regarding the MongoDB requirement. Considering that the SSPL is categorized as a non-free (proprietary) license, and this particular database is the sole supported option, it would indeed necessitate the use of the proprietary dependency flag.
> 
> Especially since other Software which uses this licence is also listed under the non-free page. But I also agree that this should be something that nodiscc should double check on.
> 
> P.S.: If I have understood the license requirements correctly, doesn't the SSPL license primarily impose restrictions on cloud service providers, owing to its extended copyleft provisions, not on average users who use this application (i.e. self-hosters)?

I would say you are correct, provided with this FAQ: https://www.mongodb.com/licensing/server-side-public-license/faq, other then that, it's hard to tell.

**Comment by nodiscc** [comment]
> MongoDB is required to use, and currently I don't have plans for people to use other database.

Then it must be moved to `non-free.md` because it requires installing non-Free software [[1]](https://github.com/awesome-selfhosted/awesome-selfhosted/issues/2689). In addition, README.md states `All rights reserved` which not appropriate for a project licensed under the GPL [[2]](https://lists.debian.org/debian-legal/2007/06/msg00252.html)

But anyway:

>  Any software project you are adding has working installation instructions.

I see no documentation about mongodb in the github project or documentation site, no mention of how to configure the application/dabatase address/credentials, the page about updating the application is empty...

So while I'm in favor of closing this PR and waiting for the project to become more documented/mature/solid, it can be moved to `non-free.md` instead - because we don't actively check that programs there have working installation instructions (and I don't personally care very much about this page as long as the formatting is OK).

**Review by nodiscc — CHANGES_REQUESTED** [review]
Please move this entry to `non-free.md`

**Comment by kokomo123** [comment]
I can agree, please move this to non-free. Thanks!

**Comment by DaneeSkripter** [comment]
Hey, I removed DaneeCloud from free self-hosting software, but **I can't move it into non-free because in non-free.md there isn't category called File Transfer - Object Storage & File Servers**

**Comment by Rabenherz112** [comment]
Feel free just to create the category for this. While we require 3 Projects to add a category to the main (free) list, you can add an already existing category from this list to the non-free list. As a reference, feel free to take a look at https://github.com/awesome-selfhosted/awesome-selfhosted/commit/bf88afade784b95bc82ae068b56f91ee567f132e.

**Comment by DaneeSkripter** [comment]
@nodiscc DaneeCloud has been moved to non-free.md

**Comment by DaneeSkripter** [comment]
@nodiscc @kokomo123 Hello?

**Comment by kokomo123** [comment]
> @nodiscc @kokomo123 Hello?

Hi, nodiscc is busy with migrating to a new repository and may be infrequent replies, but im sure nodiscc can merge this soon. Thanks for checking in.

**Comment by kokomo123** [comment]
I unfortunately cannot merge any PRs, so this will be on hold until nodiscc can be able to merge.

File: non-free.md
Comment (nodiscc) [inline non-free.md:110]:
```suggestion
- [DaneeCloud](https://docs.daneeskripter.dev/daneecloud/) - Multi-user file storage and sharing application. ([Demo](https://cloud.daneeskripter.dev), [Source Code](https://github.com/DaneeSkripter/DaneeCloud)) `GPL-3.0/SSPL-1.0` `Nodejs/JavaScript/EJS`
```

Edited the license to reflect that it requires MongoDB which is under SSPL-1.0 (this is the reason why it was moved to non-free). Also cleaner description.

**Review by nodiscc — CHANGES_REQUESTED** [review]

**Review by nodiscc — APPROVED** [review]

**Pull request merged**
